pymc_bart/utils.py

```diff
@@ -50,40 +64,46 @@ def _sample_posterior(all_trees, X, rng, size=None, excluded=None):
     for ind, p in enumerate(pred):
         for tree in stacked_trees[idx[ind]]:
             p += np.vstack([tree.predict(x, excluded) for x in X])
-    pred.reshape((*size, shape, -1))
+    pred.reshape((*size_iter, shape, -1))
     return pred
 
 
-def plot_convergence(idata, var_name=None, kind="ecdf", figsize=None, ax=None):
+def plot_convergence(
+    idata: az.InferenceData,
+    var_name: Optional[str] = None,
+    kind: str = "ecdf",
+    figsize=Tuple[float, float],
+    ax=None,
+) -> List[plt.Axes]:
     """
     Plot convergence diagnostics.
 
     Parameters
     ----------
     idata : InferenceData
         InferenceData object containing the posterior samples.
-    var_name : str
+    var_name : Optional[str]
         Name of the BART variable to plot. Defaults to None.
     kind : str
         Type of plot to display. Options are "ecdf" (default) and "kde".
-    figsize : tuple
+    figsize : Tuple[float, float]
         Figure size. Defaults to None.
     ax : matplotlib axes
         Axes on which to plot. Defaults to None.
 
     Returns
     -------
-    ax : matplotlib axes
+    List[ax] : matplotlib axes
     """
-    ess_threshold = idata.posterior.chain.size * 100
+    ess_threshold = idata["posterior"]["chain"].size * 100
```

Comment:
why this?

Reply:
Unfortunately, `mypy` does not like `idata.posterior.chain.size` since `az.InferenceData` does not always have the `.posterior` sttribute
```python
❯ mypy
pymc_bart/utils.py:98: error: "InferenceData" has no attribute "posterior"  [attr-defined]
```